File: backend/routes/reports.py
from flask import Blueprint, jsonify, request
from db import get_connection


reports_bp = Blueprint(
    "reports",
    __name__
)


# ==================================================
# Helpers
# ==================================================

def get_quarter(month):

    if month in (1, 2, 3):
        return 1

    if month in (4, 5, 6):
        return 2

    if month in (7, 8, 9):
        return 3

    return 4


def get_quarter_start(month):

    quarter = get_quarter(month)

    return ((quarter - 1) * 3) + 1


def get_quarter_end(month):

    return get_quarter_start(month) + 2


def is_quarter_end(month):

    return month in (3, 6, 9, 12)


# ==================================================
# MONTHLY / QUARTERLY VERIFICATION REPORT
# ==================================================

@reports_bp.route(
    "/monthly-verification-report",
    methods=["GET"]
)
def monthly_verification_report():

    conn = None
    cursor = None

    try:

        # ==================================================
        # INPUT
        # ==================================================

        month = request.args.get(
            "month",
            type=int
        )

        year = request.args.get(
            "year",
            type=int
        )

        if month is None or year is None:

            return jsonify({
                "message":
                    "Month and Year are required."
            }), 400

        if month < 1 or month > 12:

            return jsonify({
                "message":
                    "Invalid month."
            }), 400

        # ==================================================
        # QUARTER INFORMATION
        # ==================================================

        quarter = get_quarter(month)

        quarter_start = get_quarter_start(month)

        quarter_end = get_quarter_end(month)

        quarter_ended = is_quarter_end(month)

        # ==================================================
        # DATABASE
        # ==================================================

        conn = get_connection()

        cursor = conn.cursor(
            dictionary=True
        )

        # ==================================================
        # SELECTED MONTH CLOSURE
        #
        # Used for Rice / Wheat.
        # ==================================================

        cursor.execute("""
            SELECT
                closure_id,
                verified,
                verified_by,
                verified_at
            FROM monthly_closure
            WHERE month = %s
            AND year = %s
        """, (
            month,
            year
        ))

        monthly_closure = cursor.fetchone()

        # ==================================================
        # QUARTER-END CLOSURE
        #
        # Used for Sugar / Kerosene.
        #
        # Example:
        #
        # January / February / March
        #                    ↓
        #                  March
        #
        # April / May / June
        #              ↓
        #             June
        #
        # July / August / September
        #                   ↓
        #                September
        #
        # October / November / December
        #                    ↓
        #                 December
        # ==================================================

        cursor.execute("""
            SELECT
                closure_id,
                verified,
                verified_by,
                verified_at
            FROM monthly_closure
            WHERE month = %s
            AND year = %s
        """, (
            quarter_end,
            year
        ))

        quarter_closure = cursor.fetchone()

        # ==================================================
        # LOAD ENTITLEMENT RULES
        #
        # IMPORTANT:
        #
        # Entitlement is calculated CARD-WISE.
        #
        # PERSON:
        #     base quantity × family members
        #
        # HOUSEHOLD:
        #     base quantity
        #
        # Rice / Wheat:
        #     MONTHLY
        #
        # Sugar / Kerosene:
        #     QUARTERLY
        # ==================================================

        cursor.execute("""
            SELECT

                b.beneficiary_id,
                b.family_members,

                r.item_id,
                r.monthly_quantity,
                r.entitlement_type,
                r.entitlement_period,

                ri.item_name,
                ri.unit

            FROM beneficiaries b

            JOIN entitlement_rules r
                ON b.category_id =
                   r.category_id

            JOIN ration_items ri
                ON r.item_id =
                   ri.item_id

            WHERE LOWER(ri.item_name) IN (
                'rice',
                'wheat',
                'sugar',
                'kerosene'
            )

            ORDER BY
                ri.item_name,
                b.beneficiary_id

        """)

        entitlement_rows = cursor.fetchall()

        # ==================================================
        # REPORT ITEM STORAGE
        # ==================================================

        report_items = {}

        # ==================================================
        # PROCESS EACH CARD + ITEM
        # ==================================================

        for row in entitlement_rows:

            item_id = row["item_id"]

            item_name = (
                row["item_name"] or ""
            ).strip()

            item_key = item_name.lower()

            unit = (
                row["unit"] or "kg"
            )

            entitlement_period = (
                row["entitlement_period"]
                or "MONTHLY"
            ).upper()

            entitlement_type = (
                row["entitlement_type"]
                or "HOUSEHOLD"
            ).upper()

            base_quantity = float(
                row["monthly_quantity"] or 0
            )

            family_members = int(
                row["family_members"] or 1
            )

            # ==================================================
            # CARD-WISE ENTITLEMENT
            # ==================================================

            if entitlement_type == "PERSON":

                card_entitlement = (
                    base_quantity *
                    family_members
                )

            else:

                # HOUSEHOLD

                card_entitlement = (
                    base_quantity
                )

            # ==================================================
            # CREATE ITEM RECORD
            # ==================================================

            if item_key not in report_items:

                report_items[item_key] = {

                    "item_id":
                        item_id,

                    "item_name":
                        item_name,

                    "unit":
                        unit,

                    "period":
                        entitlement_period,

                    "total_entitled":
                        0.0,

                    "total_claimed":
                        0.0,

                    "total_unclaimed":
                        0.0,

                    "total_returned":
                        0.0,

                    "status":
                        "NOT VERIFIED"

                }

            # ==================================================
            # ADD CARD ENTITLEMENT
            # ==================================================

            report_items[
                item_key
            ]["total_entitled"] += (
                card_entitlement
            )

            # ==================================================
            # CLAIMED QUANTITY
            # ==================================================

            if entitlement_period == "QUARTERLY":

                # --------------------------------------------------
                # Sugar / Kerosene
                #
                # Claims from the ENTIRE quarter.
                # --------------------------------------------------

                cursor.execute("""
                    SELECT
                        IFNULL(
                            SUM(quantity_given),
                            0
                        ) AS claimed

                    FROM distributions

                    WHERE beneficiary_id = %s

                    AND item_id = %s

                    AND YEAR(distribution_date) = %s

                    AND MONTH(distribution_date)
                        BETWEEN %s AND %s

                """, (

                    row["beneficiary_id"],

                    item_id,

                    year,

                    quarter_start,

                    quarter_end

                ))

            else:

                # --------------------------------------------------
                # Rice / Wheat
                #
                # Claims from selected month only.
                # --------------------------------------------------

                cursor.execute("""
                    SELECT
                        IFNULL(
                            SUM(quantity_given),
                            0
                        ) AS claimed

                    FROM distributions

                    WHERE beneficiary_id = %s

                    AND item_id = %s

                    AND MONTH(distribution_date) = %s

                    AND YEAR(distribution_date) = %s

                """, (

                    row["beneficiary_id"],

                    item_id,

                    month,

                    year

                ))

            claimed_result = cursor.fetchone()

            claimed = float(
                claimed_result["claimed"] or 0
            )

            # ==================================================
            # UNCLAIMED
            # ==================================================

            unclaimed = (
                card_entitlement -
                claimed
            )

            if unclaimed < 0:

                unclaimed = 0.0

            # ==================================================
            # RETURNED
            # ==================================================

            returned = 0.0

            if entitlement_period == "QUARTERLY":

                # --------------------------------------------------
                # Sugar / Kerosene
                #
                # Returned ONLY from quarter-end audit.
                #
                # Example:
                # Q4 → December audit
                # --------------------------------------------------

                cursor.execute("""
                    SELECT
                        IFNULL(
                            SUM(
                                warehouse_returned_quantity
                            ),
                            0
                        ) AS returned

                    FROM unclaimed_audit

                    WHERE beneficiary_id = %s

                    AND item_id = %s

                    AND month = %s

                    AND year = %s

                    AND returned_to_warehouse = TRUE

                    AND audit_status = 'Returned'

                """, (

                    row["beneficiary_id"],

                    item_id,

                    quarter_end,

                    year

                ))

            else:

                # --------------------------------------------------
                # Rice / Wheat
                #
                # Returned from selected monthly audit.
                # --------------------------------------------------

                cursor.execute("""
                    SELECT
                        IFNULL(
                            SUM(
                                warehouse_returned_quantity
                            ),
                            0
                        ) AS returned

                    FROM unclaimed_audit

                    WHERE beneficiary_id = %s

                    AND item_id = %s

                    AND month = %s

                    AND year = %s

                    AND returned_to_warehouse = TRUE

                    AND audit_status = 'Returned'

                """, (

                    row["beneficiary_id"],

                    item_id,

                    month,

                    year

                ))

            returned_result = cursor.fetchone()

            returned = float(
                returned_result["returned"] or 0
            )

            # ==================================================
            # ADD CLAIMED / UNCLAIMED / RETURNED
            # ==================================================

            report_items[
                item_key
            ]["total_claimed"] += (
                claimed
            )

            report_items[
                item_key
            ]["total_unclaimed"] += (
                unclaimed
            )

            report_items[
                item_key
            ]["total_returned"] += (
                returned
            )

        # ==================================================
        # DETERMINE ITEM STATUS
        # ==================================================

        for item_key, item in report_items.items():

            period = (
                item["period"] or "MONTHLY"
            ).upper()

            total_unclaimed = float(
                item["total_unclaimed"]
            )

            total_returned = float(
                item["total_returned"]
            )

            # ==================================================
            # MONTHLY ITEMS
            #
            # Rice / Wheat
            # ==================================================

            if period == "MONTHLY":

                # --------------------------------------------------
                # Month verified
                # --------------------------------------------------

                if (
                    monthly_closure
                    and monthly_closure["verified"]
                ):

                    item["status"] = (
                        "VERIFIED"
                    )

                # --------------------------------------------------
                # Nothing to return
                # --------------------------------------------------

                elif total_unclaimed <= 0:

                    item["status"] = (
                        "NO RETURN REQUIRED"
                    )

                # --------------------------------------------------
                # All unclaimed stock returned
                # --------------------------------------------------

                elif abs(
                    total_unclaimed -
                    total_returned
                ) < 0.0001:

                    item["status"] = (
                        "READY FOR VERIFICATION"
                    )

                # --------------------------------------------------
                # Return still pending
                # --------------------------------------------------

                else:

                    item["status"] = (
                        "PENDING"
                    )

            # ==================================================
            # QUARTERLY ITEMS
            #
            # Sugar / Kerosene
            # ==================================================

            else:

                # --------------------------------------------------
                # Quarter has NOT ended.
                #
                # No warehouse return is expected yet.
                # --------------------------------------------------

                if not quarter_ended:

                    item["status"] = (
                        "QUARTER OPEN"
                    )

                # --------------------------------------------------
                # Quarter ended and verified.
                # --------------------------------------------------

                elif (
                    quarter_closure
                    and quarter_closure["verified"]
                ):

                    item["status"] = (
                        "VERIFIED"
                    )

                # --------------------------------------------------
                # No unclaimed stock.
                # --------------------------------------------------

                elif total_unclaimed <= 0:

                    item["status"] = (
                        "NO RETURN REQUIRED"
                    )

                # --------------------------------------------------
                # All unclaimed stock returned.
                # --------------------------------------------------

                elif abs(
                    total_unclaimed -
                    total_returned
                ) < 0.0001:

                    item["status"] = (
                        "READY FOR VERIFICATION"
                    )

                # --------------------------------------------------
                # Return still pending.
                # --------------------------------------------------

                else:

                    item["status"] = (
                        "PENDING"
                    )

        # ==================================================
        # ORDER ITEMS
        #
        # Rice
        # Wheat
        # Sugar
        # Kerosene
        # ==================================================

        order = [
            "rice",
            "wheat",
            "sugar",
            "kerosene"
        ]

        ordered_items = []

        for item_name in order:

            if item_name in report_items:

                ordered_items.append(
                    report_items[item_name]
                )

        # ==================================================
        # UNIT HELPERS
        # ==================================================

        def is_litre_unit(unit):

            if not unit:
                return False

            return str(unit).strip().lower() in (
                "l",
                "litre",
                "liter",
                "litres",
                "liters"
            )

        # ==================================================
        # TOTALS
        #
        # IMPORTANT:
        #
        # kg and litres must NOT be combined.
        #
        # Therefore these totals contain only
        # kilogram-based items.
        # ==================================================

        total_entitled = 0.0
        total_claimed = 0.0
        total_unclaimed = 0.0
        total_returned = 0.0

        for item in ordered_items:

            if not is_litre_unit(
                item["unit"]
            ):

                total_entitled += float(
                    item["total_entitled"]
                )

                total_claimed += float(
                    item["total_claimed"]
                )

                total_unclaimed += float(
                    item["total_unclaimed"]
                )

                total_returned += float(
                    item["total_returned"]
                )

        # ==================================================
        # LITRE TOTALS
        #
        # Separate totals for kerosene if its
        # unit is litre.
        # ==================================================

        total_entitled_litres = 0.0
        total_claimed_litres = 0.0
        total_unclaimed_litres = 0.0
        total_returned_litres = 0.0

        for item in ordered_items:

            if is_litre_unit(
                item["unit"]
            ):

                total_entitled_litres += float(
                    item["total_entitled"]
                )

                total_claimed_litres += float(
                    item["total_claimed"]
                )

                total_unclaimed_litres += float(
                    item["total_unclaimed"]
                )

                total_returned_litres += float(
                    item["total_returned"]
                )

        # ==================================================
        # OVERALL VERIFICATION STATUS
        # ==================================================

        monthly_items = [

            item

            for item in ordered_items

            if item["period"].upper()
            == "MONTHLY"

        ]

        quarterly_items = [

            item

            for item in ordered_items

            if item["period"].upper()
            == "QUARTERLY"

        ]

        # ==================================================
        # MONTHLY VERIFICATION
        # ==================================================

        monthly_verified = True

        if monthly_items:

            for item in monthly_items:

                if item["status"] not in (
                    "VERIFIED",
                    "NO RETURN REQUIRED"
                ):

                    monthly_verified = False

        # ==================================================
        # QUARTERLY VERIFICATION
        # ==================================================

        quarterly_verified = True

        if quarterly_items:

            if quarter_ended:

                for item in quarterly_items:

                    if item["status"] not in (
                        "VERIFIED",
                        "NO RETURN REQUIRED"
                    ):

                        quarterly_verified = False

            else:

                quarterly_verified = False

        # ==================================================
        # FINAL VERIFICATION STATUS
        # ==================================================

        if (
            monthly_verified
            and quarterly_verified
            and ordered_items
        ):

            verification_status = (
                "VERIFIED"
            )

        elif (
            quarterly_items
            and not quarter_ended
        ):

            verification_status = (
                "QUARTER OPEN"
            )

        else:

            verification_status = (
                "NOT VERIFIED"
            )

        # ==================================================
        # RESPONSE
        # ==================================================

        return jsonify({

            "month":
                month,

            "year":
                year,

            "quarter":
                f"Q{quarter}",

            "quarter_start_month":
                quarter_start,

            "quarter_end_month":
                quarter_end,

            "is_quarter_end":
                quarter_ended,

            "verification_status":
                verification_status,

            # --------------------------------------------------
            # KG totals
            # --------------------------------------------------

            "total_entitled":
                total_entitled,

            "total_claimed":
                total_claimed,

            "total_unclaimed":
                total_unclaimed,

            "total_returned":
                total_returned,

            # --------------------------------------------------
            # Litre totals
            # --------------------------------------------------

            "total_entitled_litres":
                total_entitled_litres,

            "total_claimed_litres":
                total_claimed_litres,

            "total_unclaimed_litres":
                total_unclaimed_litres,

            "total_returned_litres":
                total_returned_litres,

            # --------------------------------------------------
            # Item-wise data
            # --------------------------------------------------

            "items":
                ordered_items

        }), 200

    except Exception as e:

        print(
            "MONTHLY VERIFICATION REPORT ERROR:"
        )

        print(e)

        return jsonify({

            "message":
                "Unable to generate monthly "
                "verification report.",

            "error":
                str(e)

        }), 500

    finally:

        if cursor:
            cursor.close()

        if conn:
            conn.close()

File: frontend/src/pages/Reports.js
import { useEffect, useState } from "react";
import axios from "axios";
import Layout from "../components/Layout";
import "../styles/Reports.css";


function Reports() {

    const today = new Date();

    const [month, setMonth] =
        useState(today.getMonth() + 1);

    const [year, setYear] =
        useState(today.getFullYear());


    const [verificationReport, setVerificationReport] =
        useState({
            month: today.getMonth() + 1,
            year: today.getFullYear(),

            quarter: "",
            quarter_start_month: null,
            quarter_end_month: null,
            is_quarter_end: false,

            verification_status: "NOT VERIFIED",

            total_entitled: 0,
            total_claimed: 0,
            total_unclaimed: 0,
            total_returned: 0,

            items: []
        });


    const monthNames = [
        "January",
        "February",
        "March",
        "April",
        "May",
        "June",
        "July",
        "August",
        "September",
        "October",
        "November",
        "December"
    ];


    // =========================================
    // LOAD REPORT
    // =========================================

    useEffect(() => {

        loadVerificationReport();

    }, [month, year]);


    const loadVerificationReport = async () => {

        try {

            const response = await axios.get(
                "http://127.0.0.1:5000/monthly-verification-report",
                {
                    params: {
                        month,
                        year
                    }
                }
            );


            setVerificationReport(
                response.data
            );

        }

        catch (err) {

            console.log(
                "Verification report error:",
                err
            );

            setVerificationReport({

                month,
                year,

                quarter: "",
                quarter_start_month: null,
                quarter_end_month: null,
                is_quarter_end: false,

                verification_status:
                    "NOT VERIFIED",

                total_entitled: 0,
                total_claimed: 0,
                total_unclaimed: 0,
                total_returned: 0,

                items: []

            });

        }

    };


    // =========================================
    // GET ITEM
    // =========================================

    const getItem = (name) => {

        return verificationReport.items.find(
            (item) =>
                String(item.item_name || "")
                    .trim()
                    .toLowerCase() ===
                name.toLowerCase()
        );

    };


    const rice = getItem("rice");
    const wheat = getItem("wheat");
    const sugar = getItem("sugar");
    const kerosene = getItem("kerosene");


    // =========================================
    // FORMAT QUANTITY
    // =========================================

    const formatQuantity = (
        item,
        field = "total_entitled"
    ) => {

        if (!item) {
            return "0.00";
        }

        return Number(
            item[field] || 0
        ).toFixed(2);

    };


    // =========================================
    // FORMAT UNIT
    // =========================================

    const getUnit = (item) => {

        if (!item) {
            return "kg";
        }

        const unit =
            String(item.unit || "kg")
                .trim()
                .toLowerCase();


        if (
            unit === "l" ||
            unit === "litre" ||
            unit === "liter" ||
            unit === "litres" ||
            unit === "liters"
        ) {

            return "Litre";

        }


        return "Kg";

    };


    // =========================================
    // QUARTER MESSAGE
    // =========================================

    const quarterMessage =
        verificationReport.is_quarter_end

            ?

            "Quarter has ended. Sugar and Kerosene are now eligible for warehouse return and verification."

            :

            "Quarter is still open. Sugar and Kerosene can be collected during the quarter and are audited for return only at quarter end.";


    // =========================================
    // VERIFICATION STATUS CLASS
    // =========================================

    const verificationStatusClass =
        verificationReport.verification_status ===
        "VERIFIED"

            ?

            "verified"

            :

            "not-verified";


    return (

        <Layout>

            <div className="reports-container">


                {/* =================================
                    PAGE HEADER
                ================================= */}

                <div className="reports-header">

                    <h1>
                        Reports Dashboard
                    </h1>

                    <p className="reports-subtitle">

                        View entitlement verification
                        reports.

                    </p>

                </div>



                {/* =================================
                    ENTITLEMENT VERIFICATION
                ================================= */}

                <div
                    className="report-section"
                    id="verification-report"
                >


                    {/* =================================
                        SECTION HEADER
                    ================================= */}

                    <div className="section-header">

                        <div>

                            <h2>
                                Entitlement Verification
                            </h2>

                            <p>

                                Monthly verification for
                                Rice and Wheat, and quarterly
                                verification for Sugar and
                                Kerosene.

                            </p>

                        </div>

                    </div>



                    {/* =================================
                        MONTH / YEAR FILTER
                    ================================= */}

                    <div className="report-filter">

                        <div className="filter-group">

                            <label>
                                Month
                            </label>

                            <select
                                value={month}
                                onChange={(e) =>
                                    setMonth(
                                        Number(
                                            e.target.value
                                        )
                                    )
                                }
                            >

                                {
                                    monthNames.map(
                                        (
                                            monthName,
                                            index
                                        ) => (

                                            <option
                                                key={index}
                                                value={index + 1}
                                            >

                                                {monthName}

                                            </option>

                                        )
                                    )
                                }

                            </select>

                        </div>



                        <div className="filter-group">

                            <label>
                                Year
                            </label>

                            <input
                                type="number"
                                value={year}
                                onChange={(e) =>
                                    setYear(
                                        Number(
                                            e.target.value
                                        )
                                    )
                                }
                            />

                        </div>

                    </div>



                    {/* =================================
                        VERIFICATION HEADER
                    ================================= */}

                    <div className="verification-header">


                        <div className="verification-period">

                            <span>
                                Report Period
                            </span>

                            <strong>

                                {
                                    monthNames[
                                        (
                                            verificationReport.month ||
                                            month
                                        ) - 1
                                    ]
                                }{" "}

                                {
                                    verificationReport.year ||
                                    year
                                }

                            </strong>

                        </div>



                        <div className="verification-quarter">

                            <span>
                                Quarter
                            </span>

                            <strong>

                                {
                                    verificationReport.quarter ||
                                    "-"
                                }

                            </strong>

                        </div>



                        <div className="verification-quarter-status">

                            <span>
                                Quarter Status
                            </span>

                            <strong>

                                {
                                    verificationReport.is_quarter_end
                                        ? "QUARTER END"
                                        : "QUARTER OPEN"
                                }

                            </strong>

                        </div>



                        <div className="verification-status">

                            <span>
                                Verification Status
                            </span>

                            <strong
                                className={
                                    verificationStatusClass
                                }
                            >

                                {
                                    verificationReport.verification_status ===
                                    "VERIFIED"

                                        ?

                                        "✓ VERIFIED"

                                        :

                                        "⚠ " +
                                        (
                                            verificationReport.verification_status ||
                                            "NOT VERIFIED"
                                        )
                                }

                            </strong>

                        </div>

                    </div>



                    {/* =================================
                        QUARTERLY INFORMATION
                    ================================= */}

                    <div className="quarter-message">

                        <strong>
                            Sugar & Kerosene:
                        </strong>{" "}

                        {quarterMessage}

                    </div>



                    {/* =================================
                        ENTITLEMENT SUMMARY CARDS
                    ================================= */}

                    <div className="entitlement-summary">


                        {/* Rice */}

                        <div className="entitlement-card">

                            <h3>

                                {formatQuantity(
                                    rice
                                )}{" "}

                                {getUnit(rice)}

                            </h3>

                            <p>
                                Rice Entitled
                            </p>

                        </div>



                        {/* Wheat */}

                        <div className="entitlement-card">

                            <h3>

                                {formatQuantity(
                                    wheat
                                )}{" "}

                                {getUnit(wheat)}

                            </h3>

                            <p>
                                Wheat Entitled
                            </p>

                        </div>



                        {/* Sugar */}

                        <div className="entitlement-card">

                            <h3>

                                {formatQuantity(
                                    sugar
                                )}{" "}

                                {getUnit(sugar)}

                            </h3>

                            <p>
                                Sugar Qtr. Entitlement
                            </p>

                        </div>



                        {/* Kerosene */}

                        <div className="entitlement-card">

                            <h3>

                                {formatQuantity(
                                    kerosene
                                )}{" "}

                                {getUnit(kerosene)}

                            </h3>

                            <p>
                                Kerosene Qtr. Entitlement
                            </p>

                        </div>

                    </div>



                    {/* =================================
                        VERIFICATION TABLE
                    ================================= */}

                    <div className="table-wrapper">

                        <table>

                            <thead>

                                <tr>

                                    <th>
                                        Item
                                    </th>

                                    <th>
                                        Period
                                    </th>

                                    <th>
                                        Total Entitled
                                    </th>

                                    <th>
                                        Total Claimed
                                    </th>

                                    <th>
                                        Unclaimed
                                    </th>

                                    <th>
                                        Returned
                                    </th>

                                    <th>
                                        Status
                                    </th>

                                </tr>

                            </thead>



                            <tbody>

                                {
                                    verificationReport.items.length > 0

                                        ?

                                        verificationReport.items.map(
                                            (
                                                item,
                                                index
                                            ) => (

                                                <tr
                                                    key={index}
                                                >

                                                    <td>

                                                        <strong>
                                                            {
                                                                item.item_name
                                                            }
                                                        </strong>

                                                    </td>



                                                    <td>

                                                        {
                                                            item.period
                                                        }

                                                    </td>



                                                    <td>

                                                        {
                                                            Number(
                                                                item.total_entitled ||
                                                                0
                                                            ).toFixed(2)
                                                        }{" "}

                                                        {
                                                            getUnit(item)
                                                        }

                                                    </td>



                                                    <td>

                                                        {
                                                            Number(
                                                                item.total_claimed ||
                                                                0
                                                            ).toFixed(2)
                                                        }{" "}

                                                        {
                                                            getUnit(item)
                                                        }

                                                    </td>



                                                    <td>

                                                        {
                                                            Number(
                                                                item.total_unclaimed ||
                                                                0
                                                            ).toFixed(2)
                                                        }{" "}

                                                        {
                                                            getUnit(item)
                                                        }

                                                    </td>



                                                    <td>

                                                        {
                                                            Number(
                                                                item.total_returned ||
                                                                0
                                                            ).toFixed(2)
                                                        }{" "}

                                                        {
                                                            getUnit(item)
                                                        }

                                                    </td>



                                                    <td>

                                                        <span
                                                            className={
                                                                item.status ===
                                                                "VERIFIED"

                                                                    ?

                                                                    "table-status verified"

                                                                    :

                                                                    "table-status not-verified"
                                                            }
                                                        >

                                                            {
                                                                item.status ===
                                                                "VERIFIED"

                                                                    ?

                                                                    "✓ Verified"

                                                                    :

                                                                    item.status
                                                            }

                                                        </span>

                                                    </td>

                                                </tr>

                                            )
                                        )

                                        :

                                        <tr>

                                            <td
                                                colSpan="7"
                                                className="empty-state"
                                            >

                                                No entitlement
                                                records were found
                                                for this month.

                                            </td>

                                        </tr>

                                }

                            </tbody>

                        </table>

                    </div>



                    {/* =================================
                        INFORMATION FOOTER
                    ================================= */}

                    <div className="report-information">

                        <div>

                            <strong>
                                Inventory Update
                            </strong>

                            <span>
                                Returned stock is added back
                                to inventory only after
                                warehouse manager verification.
                            </span>

                        </div>



                        <div>

                            <strong>
                                Rice / Wheat
                            </strong>

                            <span>
                                Audited and returned at the
                                end of every month.
                            </span>

                        </div>



                        <div>

                            <strong>
                                Sugar / Kerosene
                            </strong>

                            <span>
                                Audited and returned only
                                at quarter end.
                            </span>

                        </div>

                    </div>

                </div>



                {/* =================================
                    FOOTER
                ================================= */}

                <div className="report-footer">

                    Report generated on{" "}

                    {
                        new Date().toLocaleString()
                    }

                </div>

            </div>

        </Layout>

    );

}


export default Reports;